flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV120.java

```diff
@@ -162,15 +174,23 @@ public CatalogColumnStatisticsDataDate toFlinkDateColStats(ColumnStatisticsData
 		try {
 			Object dateStats = ColumnStatisticsData.class.getDeclaredMethod("getDateStats").invoke(hiveDateColStats);
 			Class dateStatsClz = dateStats.getClass();
-			long numDV = (long) dateStatsClz.getMethod("getNumDVs").invoke(dateStats);
-			long numNull = (long) dateStatsClz.getMethod("getNumNulls").invoke(dateStats);
+			boolean isSetNumDv = (boolean) dateStatsClz.getMethod("isSetNumDVs").invoke(dateStats);
+			boolean isSetNumNull = (boolean) dateStatsClz.getMethod("isSetNumNulls").invoke(dateStats);
+			boolean isSetHighValue = (boolean) dateStatsClz.getMethod("isSetHighValue").invoke(dateStats);
+			boolean isSetLowValue = (boolean) dateStatsClz.getMethod("isSetLowValue").invoke(dateStats);
+			Long numDV = isSetNumDv ? (Long) dateStatsClz.getMethod("getNumDVs").invoke(dateStats) : null;
+			Long numNull = isSetNumNull ? (Long) dateStatsClz.getMethod("getNumNulls").invoke(dateStats) : null;
 			Object hmsHighDate = dateStatsClz.getMethod("getHighValue").invoke(dateStats);
 			Object hmsLowDate = dateStatsClz.getMethod("getLowValue").invoke(dateStats);
 			Class hmsDateClz = hmsHighDate.getClass();
 			Method hmsDateDays = hmsDateClz.getMethod("getDaysSinceEpoch");
-			long highDateDays = (long) hmsDateDays.invoke(hmsHighDate);
-			long lowDateDays = (long) hmsDateDays.invoke(hmsLowDate);
-			return new CatalogColumnStatisticsDataDate(new Date(lowDateDays), new Date(highDateDays), numDV, numNull);
+			Long highDateDays = isSetHighValue ? (Long) hmsDateDays.invoke(hmsHighDate) : null;
+			Long lowDateDays = isSetLowValue ? (Long) hmsDateDays.invoke(hmsLowDate) : null;
+			return new CatalogColumnStatisticsDataDate(
+					isSetLowValue ? new Date(lowDateDays) : null,
```

Comment:
Could you group the logical related to `isSetLowValue` or `isSetHighValue` together? Currently this flag is used in different places but with the same purpose.